python/src/helpers/count_matrix.py

```diff
@@ -22,84 +27,57 @@ def get_objects(self):
         if not objects:
             return {}
 
-        objects = {o["Key"]: o["ETag"] for o in objects if o["Size"] > 0}
+        objects = {o["Key"]: o["LastModified"] for o in objects if o["Size"] > 0}
 
         return objects
 
-    def md5_full_file(self, file_path):
-        hash_md5 = hashlib.md5()
-        with open(file_path, "rb") as f:
-            for chunk in iter(lambda: f.read(4096), b""):
-                hash_md5.update(chunk)
-        return hash_md5.hexdigest()
-
-    def megabyte_chunks(self, filesize, num_parts):
-        x = filesize / int(num_parts)
-        y = x % 1048576
-        return int(x + 1048576 - y)
-
-    def validate_etag(self, file_path, etag):
-
-        # get the size of the file. if it doesn't exist,
-        # the etags definitely don't match
-        try:
-            filesize = os.path.getsize(file_path)
-        except OSError:
-            return False
+    @xray_recorder.capture("CountMatrix.download_object")
+    def download_object(self, key, last_modified):
+        path = os.path.join(config.LOCAL_DIR, key)
+        print(key)
 
-        # get the number of parts. if this doesn't exist,
-        # the etag has no parts, so it is just the md5 digest
-        etag = etag.replace('"', "")
+        last_mod_local = None
 
         try:
-            num_parts = int(etag.split("-")[1])
-        except IndexError:
-            return etag == self.md5_full_file(file_path)
-
-        # these are some common part sizes used. the last one
-        # is the file size divided by the number of parts,
-        # rounded to the nearest megabyte.
-        partsizes = [
-            8388608,
-            15728640,
-            16777216,
-            self.megabyte_chunks(filesize, num_parts),
-        ]
-
-        valid_sizes = lambda partsize: (
-            partsize < filesize and (float(filesize) / float(partsize)) <= num_parts
-        )
-
-        for partsize in filter(valid_sizes, partsizes):
-            md5_digests = []
-
-            with open(file_path, "rb") as f:
-                for chunk in iter(lambda: f.read(partsize), b""):
-                    md5_digests.append(hashlib.md5(chunk).digest())
-
-            candidate = (
-                hashlib.md5(b"".join(md5_digests)).hexdigest()
-                + "-"
-                + str(len(md5_digests))
+            last_mod_local = datetime.datetime.fromtimestamp(
+                os.path.getmtime(path), tz=timezone.utc
             )
+        except Exception as e:
+            print(e)
```

Comment:
should this print be added to current segment? current_segment().add_exception()

Reply:
No, this will only occur if the file does not exist or there was some kind of error getting the last modified date. We don't need to throw anything here, we can just re-download the file.